app/src/main/java/com/p2p/presentation/truco/actions/TrucoActionsBottomSheetFragment.kt

```diff
@@ -105,18 +148,36 @@ class TrucoActionsBottomSheetFragment : BaseBottomSheetDialogFragment<ViewTrucoA
 
     private fun onCollapsed() {
         binding.openButtonIcon.animateRotation(0f)
-        changeVisibleButtonsEnable(false)
+        disableEnvidoAndTrucoButtons()
     }
 
     private fun onExpanded() {
         binding.openButtonIcon.animateRotation(180f)
-        changeVisibleButtonsEnable(true)
+        enableEnvidoAndTrucoButtons()
+    }
+
+    private fun disableEnvidoAndTrucoButtons() {
+        changeEnvidoButtonAvailability(false)
+        changeTrucoButtonAvailability(false)
+    }
+
+    private fun enableEnvidoAndTrucoButtons() {
+        changeEnvidoButtonAvailability(true)
+        changeTrucoButtonAvailability(true)
+    }
+
+    private fun disableEnvidoForHand() {
+        changeEnvidoButtonAvailability(false)
+        envidoDisabledForHand = true
+    }
+
+    private fun changeEnvidoButtonAvailability(enabled: Boolean) {
+        binding.envidoOptionsButton.isEnabled = enabled && !envidoDisabledForHand
     }
 
-    private fun changeVisibleButtonsEnable(isEnabled: Boolean) {
-        binding.envidoOptionsButton.isEnabled = isEnabled
-        binding.trucoButton.isEnabled = isEnabled
-        binding.trucoButton.alpha = if (isEnabled) 1f else 0.3f
+    private fun changeTrucoButtonAvailability(enabled: Boolean) {
+        binding.trucoButton.isEnabled = (enabled && !trucoDisabledForHand)
```

Comment:
Me parece que terminan dando muchas vueltas con esto por lo que ya existía y termina quedando complejo cuando es algo sencillo, yo volvería a lo anterior pero con un cambio

```suggestion
    private fun changeButtonsEnable(isEnvidoEnabled: Boolean, isTrucoEnabled: Boolean) {
        binding.envidoOptionsButton.isEnabled = isEnvidoEnabled
        binding.trucoButton.isEnabled = isTrucoEnabled```

Reply:
es que no, faltaría otro booleano para ver si está deshabilitado por toda la mano, no alcanza con el que se manda por el hecho de abrir o cerrar el bottomSheet

Reply:
Pero es que justamente, lo que se hace es guardarse esas variables y esto habilitarlo cuando corresponda 🤔